snaps_k8s/provision/ansible_configuration.py

```diff
@@ -232,7 +232,27 @@ def __kubespray(k8s_conf, base_pb_vars):
             consts.K8_CPU_PINNING_CONFIG,
             variables={'KUBESPRAY_PATH': config_utils.get_kubespray_dir(
                 k8s_conf)})
+                
+    ha_enabled = False
+    ha_configuration = config_utils.get_ha_config(k8s_conf)
+    if ha_configuration:
+        for ha_config_list_data in ha_configuration:
+            lb_ip = ha_config_list_data.get(consts.HA_API_EXT_LB_KEY).get("ip")
+
+        logger.info("INSIDE KUBESPPRAY Load balancer ip %s", lb_ip)
+        ha_enabled = True
+        pb_vars = {
+        'SRC_PACKAGE_PATH': config_utils.get_artifact_dir(k8s_conf),
+        'PROJ_ARTIFACT_DIR': config_utils.get_project_artifact_dir(k8s_conf),
+        'Project_name': config_utils.get_project_name(k8s_conf),
+        }
 
+        pb_vars.update(config_utils.get_proxy_dict(k8s_conf))
+        loadbalancer_dict = __get_loadbalancer_dict(k8s_conf)
+        if loadbalancer_dict:
+            __ha_configuration(loadbalancer_dict,pb_vars,k8s_conf,config_utils.get_project_name(k8s_conf))
+            
```

Comment:
Why not just send in the k8s_conf here and extract the loadbalancer_dict, pb_vars, and project name inside of the __ha_configuration() function?

Reply:
Incorporated.